### PR TITLE
Add more haddock and move some things around a wee bit.

### PR DESCRIPTION
**Pull request by stevana**

### CODE REVIEW

File: src/Test/StateMachine/Types.hs [hunk 1]
```diff
@@ -75,8 +75,11 @@ data Untyped (act :: (* -> *) -> * -> *) where
 
 ------------------------------------------------------------------------
 
+-- | A (non-failing) state machine record bundles up all functionality
+--   needed to perform our tests.
 type StateMachine model act m = StateMachine' model act Void m
 
+-- | Same as above, but with possibily failing semantics.
```
Comment (Lysxia) [inline src/Test/StateMachine/Types.hs]: "possibly"

File: src/Test/StateMachine/Types/History.hs [hunk 2]
```diff
@@ -3,47 +3,77 @@
...
+-- |
+-- Module      :  Test.StateMachine.Types.History
+-- Copyright   :  (C) 2017, ATS Advanced Telematic Systems GmbH
+-- License     :  BSD-style (see the file LICENSE)
+--
+-- Maintainer  :  Stevan Andjelkovic <stevan@advancedtelematic.com>
+-- Stability   :  provisional
+-- Portability :  non-portable (GHC extensions)
+--
+-- This module contains the notion of a history of an exection of a (parallel)
```
Comment (Lysxia) [inline src/Test/StateMachine/Types/History.hs]: "execution"

File: src/Test/StateMachine/Types/History.hs [hunk 3]
```diff
@@ -54,3 +84,32 @@ findCorrespondingResp _   [] = []
...
   [ (resp, e : es') | (resp, es') <- findCorrespondingResp pid es ]
+
+------------------------------------------------------------------------
+
+-- | An operation packs up an invocation event with its corresponding
+--   response event.
+data Operation act err = forall resp. Typeable resp =>
+  Operation (act Concrete resp) String (Result (Concrete resp) err) Pid
+
+-- | Given a history, return all possible interleavings of invocation
```
Comment (Lysxia) [inline src/Test/StateMachine/Types/History.hs]: "invocations"

**Comment by stevana** [comment]
Fixed, thanks!

**Pull request merged**